### PR TITLE
Make squid compile with modern compilers

### PR DESCRIPTION
**Pull request by AntonBikineev**
I just tried compiling squid and ran into compile time errors. This is PR aimed to fix them.

### CODE REVIEW

**Comment by AntonBikineev** [comment]
is there anybody checking/reviewing PRs?

**Comment by clarkli86** [comment]
I had to do the same thing to make squid compile using gcc 5.3 under Linux x64

File: src/acl/Acl.h [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #include <ostream>
 #include <string>
 #include <vector>
+#include <map>
```
Comment (rousskov) [inline src/acl/Acl.h]: Already fixed (and using proper #include order) in trunk r14486.

File: helpers/defines.h [hunk 2]
```diff
@@ -39,6 +39,14 @@
 #define __SQUID_HELPERS_DEFINES_H
 
 /*
+ * As long as every file including this header
+ * is supposed to use macros defined in it,
+ * it is reasonable to include iostream header
+ * from here
+ */
+#include <iostream>
```
Comment (rousskov) [inline helpers/defines.h]: Already fixed in trunk r14487 (the file got moved to src/helper/protocol_defines.h).

**Comment by rousskov** [comment]
Unfortunately, the Squid Project is not using git (yet?). Github activity is likely to be ignored or forgotten. The best way to get your changes accepted may be to follow the standard procedure outlined at http://wiki.squid-cache.org/MergeProcedure

**Pull request closed without merging**